src/openrct2-ui/scripting/ScUi.hpp

```diff
@@ -388,6 +406,7 @@ namespace OpenRCT2::Scripting
             dukglue_register_method(ctx, &ScUi::registerMenuItem, "registerMenuItem");
             dukglue_register_method(ctx, &ScUi::registerToolboxMenuItem, "registerToolboxMenuItem");
             dukglue_register_method(ctx, &ScUi::registerShortcut, "registerShortcut");
+            dukglue_register_method(ctx, &ScUi::setDropdownColours, "setDropdownColours");
```

Comment:
What is the use-case for this? And would it not possible to have this as a property per colour picker?

In the current state, this can mess with other plugins using the colour picker as well.

Would also need a getter for getting the current colour selection (for that colour picker).

Reply:
It's so that you can customise the list of colours in the picker. Some people would like to simplify restricting what colours they can use

Reply:
I agree that it would create a mess between different plugins. It can easily create situations where one plugin wants to have a very restricted colour picker (maybe only two colours) and the dev sets this as default for all colour pickers because he was not aware of the implications.
How difficult would it be to have this per widget instead of global?

Reply:
Would require breaking the api of the custom window code. Is there anyone actually wanting to restrict per custom window widget colour pickers. Would be a bit odd because you would still need this global setter but would need to know it only affects non plugin windows.

Reply:
Actually I do not see why you would need this setter at all. Why would you want to restrict all colour pickers in the game to a certain set of colours?
On the other hand, I can probably think of use cases where you want specific colour pickers to have restricted options.

Reply:
Okay, I can see that, but why should it be part of the plugin API and not somewhere in the settings?

Reply:
What will happen if two plugins call this function? Will they just compete and the last caller wins? What happens if a plugin sets the currently selected colour of the picker to a colour that is not in the list of available colours?

Reply:
> What will happen if two plugins call this function? Will they just compete and the last caller wins? What happens if a plugin sets the currently selected colour of a specific picker to a colour that is not in the list of available colours?

Last caller wins. If you have selected something that isn't in the list prior (or it's a default colour for the window) then that selection stays the same. This only affects newly opened drop downs. This allows the pipette tool to work as well.